src/helpers/files.ts

```diff
@@ -293,18 +294,15 @@ export function readAllLines(filePath: string): string[] {
  * @param {string} filePath
  * @returns {string}
  */
-export function readCoverageFile(
+export async function readCoverageFile(
   projectRoot: string,
   filePath: string,
-): string {
-  try {
-    return fs.readFileSync(getFilePath(projectRoot, filePath), {
-      encoding: 'utf-8',
-    })
-  } catch (error) {
-    logAndThrow(`There was an error reading the coverage file: ${error}`)
-  }
-  return ''
+): Promise<string> {
+  return readFile(getFilePath(projectRoot, filePath), {
+    encoding: 'utf-8',
+  }).catch(err => {
+    logAndThrow(`There was an error reading the coverage file: ${err}`)
+  })
```

Comment:
```suggestion
  })
```

Reply:
Why?

Reply:
It's the caller's responsibility to handle errors / catch and log them. This is actually what appears to be happening right now so the change would not introduce incompatibility of anything that calls it:

https://github.com/codecov/uploader/blob/7567c5e5c6586ad9ddf151b1807c3bba629daca3/src/index.ts#L182

Reply:
Additionally, the catch-and-throw pattern, when used this way, creates nested errors that are wrapped several times (which is why we're seeing "Error: Received an Error: Received an Error: Received an unexpected response" and the like).

Reply:
Ah. Sorry I didn't see your response before merge